app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,37 @@
+class PasswordResetsController < ApplicationController
+  layout 'static_pages'
+
+  def new
+  end
+
+  def create
+    user = User.find_by_email(params[:email])
+    user.send_password_reset if user
+    flash[:success] = "Check your email for password reset instructions, then hurry back here for joyful fun times!"
+    redirect_to root_url
+  end
+
+  def edit
+    @user = User.find_by_password_reset_token!(params[:id])
+  end
+
+  def update
```

Comment:
Assignment Branch Condition size for update is too high. [21.12/15]<br>Method has too many lines. [11/10]